src/pages/projectSettings/tabs/GeneralTab/GeneralTab.tsx

```diff
@@ -67,6 +67,21 @@ export const GeneralTab: React.VFC<TabProps> = ({
       onChange={onChange}
       schema={schema}
       uiSchema={uiSchema}
+      validate={validate as any}
     />
   );
 };
+
+const validate = ((formData, errors) => {
+  const {
+    generalConfiguration: { enabled, branch },
+  } = formData;
+
+  if (enabled && !branch) {
+    errors.generalConfiguration.branch.addError(
+      "A branch is required for enabled projects."
+    );
+  }
```

Comment:
works good for me but I just had one question! should we prevent saving if it has a default from repo branch?

<img width="475" alt="Screenshot 2023-04-27 at 2 53 39 PM" src="https://user-images.githubusercontent.com/47064971/234963828-a7052372-f841-41e9-af55-9aa8502200f0.png">